app/models/sharing_strategies/project_query_strategy.rb

```diff
@@ -88,6 +88,21 @@ def shares(reload: false)
       end
     end
 
+    def share_description(share) # rubocop:disable Metrics/AbcSize,Metrics/PerceivedComplexity
+      return I18n.t("sharing.user_details.invited") if !manageable? && share.principal.invited?
+      return "" if !manageable?
+
+      if share.principal == entity.user
+        I18n.t("sharing.project_queries.user_details.owner")
+      elsif entity.public?
+        if share.principal.is_a?(User) && share.principal.allowed_globally?(:manage_public_project_queries)
+          I18n.t("sharing.project_queries.user_details.can_manage_public_lists")
+        elsif share.roles.any? { |role| role.builtin == Role::BUILTIN_PROJECT_QUERY_VIEW }
+          I18n.t("sharing.project_queries.user_details.can_view_because_public")
+        end
+      end
```

Comment:
`I18n.t("sharing.project_queries.user_details.…")` can be extracted

Reply:
Same argument as the other case. It leads to less readable code if we somehow set a variable. It's already complicated enough as it is.

Reply:
In this case it is much less code, but still to me extracting variable to remove repetition sounds good, the only downside I imagine is that it is not possible to find occurrence by complete key.

Also either this should explicitly return `""` in else cases or the guard on the second line should return nil (explicitly or implicitly).